library/sinks/AwsSDKVersion3.ts

```diff
@@ -141,51 +142,63 @@ export class AwsSDKVersion3 implements Wrapper {
+              returnValue
+                .then((response) => {
+                  if (
+                    exports.InvokeModelCommand &&
+                    command instanceof exports.InvokeModelCommand
+                  ) {
+                    this.processInvokeModelResponse(response, command, agent);
+                  } else if (
+                    exports.ConverseCommand &&
+                    command instanceof exports.ConverseCommand
+                  ) {
+                    this.processConverseResponse(response, command, agent);
+                  }
+                })
+                .catch((error) => {
+                  agent.onErrorThrownByInterceptor({
+                    error: error,
+                    method: "send.<promise>",
+                    module: "@aws-sdk/client-bedrock-runtime",
+                  });
+                });
+            }
+          }
+
+          return returnValue;
+        },
+      });
+    }
+  }
+
   wrap(hooks: Hooks) {
     hooks
       .addPackage("@aws-sdk/client-bedrock-runtime")
       .withVersion("^3.0.0")
       .onRequire((exports, pkgInfo) => {
-        if (exports.BedrockRuntimeClient) {
-          wrapExport(exports.BedrockRuntimeClient.prototype, "send", pkgInfo, {
-            kind: "ai_op",
-            modifyReturnValue: (args, returnValue, agent) => {
-              if (args.length > 0) {
-                const command = args[0];
-                if (returnValue instanceof Promise) {
-                  // Inspect the response after the promise resolves, it won't change the original promise
-                  returnValue
-                    .then((response) => {
-                      if (
-                        exports.InvokeModelCommand &&
-                        command instanceof exports.InvokeModelCommand
-                      ) {
-                        this.processInvokeModelResponse(
-                          response,
-                          command,
-                          agent
-                        );
-                      } else if (
-                        exports.ConverseCommand &&
-                        command instanceof exports.ConverseCommand
-                      ) {
-                        this.processConverseResponse(response, command, agent);
-                      }
-                    })
-                    .catch((error) => {
-                      agent.onErrorThrownByInterceptor({
-                        error: error,
-                        method: "send.<promise>",
-                        module: "@aws-sdk/client-bedrock-runtime",
-                      });
-                    });
-                }
-              }
-
-              return returnValue;
-            },
-          });
-        }
+        this.onRequire(exports, pkgInfo);
+      })
+      .addFileInstrumentation({
+        path: "dist-cjs/index.js",
```

Comment:
ESM missing? In that case we would need to modify the BedrockRuntimeClient class directly, not possible to access or modify exports. So maybe create a function that directly accepts the BedrockRuntimeClient?